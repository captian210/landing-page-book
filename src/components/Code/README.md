# Code

[![standard-readme compliant](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)

## Overview

`@front10/landing-page-book/Code` is used to write/show code in many language.

## How to use

1- Importing bootstrap style

```js
import 'bootstrap/dist/css/bootstrap.min.css';
```

2- Importing font-awesome or any other font (Optional)

```js
import 'font-awesome/css/font-awesome.min.css';
```

3- Add following content to import the component:

```js
import Code from '@front10/landing-page-book/dist/components/Code';
import '@front10/landing-page-book/dist/components/Code/style.css';
```

> Note: For including all components styles once you can use [a theme](https://github.com/front10/landing-page-book/wiki/Theming).

4- Put this code into jsx page

```html
<Code />
```

<a target="_blank" href="https://codesandbox.io/s/pmjvk5wl27">
  <img alt="Edit Code demo" src="https://codesandbox.io/static/img/play-codesandbox.svg">
</a>

## Properties

| </br>Name    | </br>Type | </br>Summary                                                                        |
| ------------ | --------- | ----------------------------------------------------------------------------------- |
| readOnly     | `Boolean` | No allow to change the code, Default `false`                                        |
| lineNumbers  | `Boolean` | Shows line numbers, Default `true`                                                  |
| showheader   | `Boolean` | Shows header bar, Default `true`                                                    |
| showfooter   | `Boolean` | Shows footer bar, Default `true`                                                    |
| code         | `String`  | Code to show on component, Default `Js code example`                                |
| theme        | `String`  | Theme of code, Default `"idea"`                                                     |
| languageCode | `String`  | language in which the code is written to embellish the syntax, Default `javascript` |
| updateCode   | `func`    | Function to handle the change made in the code.                                     |

## Theming variables

| </br>Name                               | </br>Summary                          | </br>Default     |
| --------------------------------------- | ------------------------------------- | ---------------- |
| --CodeMirror\_\_lines_backgroundColor   | Code's lines background color         | `#fff`           |
| --CodeMirror\_\_header_with             | Code's header width                   | `100%`           |
| --CodeMirror\_\_header_height           | Code's header height                  | `45px`           |
| --CodeMirror\_\_header_backgroundColor  | Code's header background color        | `#dcdcdc`        |
| --CodeMirror\_\_footer_with             | Code's footer width                   | `100%`           |
| --CodeMirror\_\_footer_height           | Code's footer height                  | `15px`           |
| --CodeMirror\_\_footer_backgroundColor  | Code's footer bacground color         | `#dcdcdc`        |
| --CodeMirror\_\_header_copybtn_margin   | Margin of copy button in header       | `10px -10px 0 0` |
| --CodeMirror\_\_header_deletebtn_margin | Margin of clear code button in header | `10px 0 0 0`     |
| --CodeMirror\_\_height                  | Height of Code                        | `250px`          |
| --CodeMirror\_\_font_family             | Code's font family                    | `Lato`           |
