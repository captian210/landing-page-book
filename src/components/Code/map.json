{"description":"","displayName":"Code","methods":[{"name":"onUpdate","docblock":null,"modifiers":[],"params":[{"name":"prevProps","type":null}],"returns":null},{"name":"updateCode","docblock":null,"modifiers":[],"params":[{"name":"newCode","type":null}],"returns":null},{"name":"clearCode","docblock":null,"modifiers":[],"params":[],"returns":null},{"name":"toggleReadOnly","docblock":null,"modifiers":[],"params":[],"returns":null},{"name":"copyToClipboard","docblock":null,"modifiers":[],"params":[],"returns":null},{"name":"toggleCollapse","docblock":null,"modifiers":[],"params":[],"returns":null}],"props":{"readOnly":{"type":{"name":"bool"},"required":false,"description":"No allow to change the code, Default `false`","defaultValue":{"value":"false","computed":false}},"collapsible":{"type":{"name":"bool"},"required":false,"description":"Define if component is collapsible, Default `false`","defaultValue":{"value":"false","computed":false}},"collapsed":{"type":{"name":"bool"},"required":false,"description":"Define if component is collapsed, Default `false`","defaultValue":{"value":"false","computed":false}},"lineNumbers":{"type":{"name":"bool"},"required":false,"description":"Shows line numbers, Default `true`","defaultValue":{"value":"true","computed":false}},"showheader":{"type":{"name":"bool"},"required":false,"description":"Shows header bar, Default `true`","defaultValue":{"value":"true","computed":false}},"showDeleteButton":{"type":{"name":"bool"},"required":false,"description":"Show or hide delete button, Default `true`","defaultValue":{"value":"true","computed":false}},"showCopyButton":{"type":{"name":"bool"},"required":false,"description":"Show or hide copy button, Default `true`","defaultValue":{"value":"true","computed":false}},"code":{"type":{"name":"string"},"required":true,"description":"Code to show on component, Default `Js code example`"},"languageCode":{"type":{"name":"string"},"required":false,"description":"language in which the code is written to embellish the syntax, Default `javascript`","defaultValue":{"value":"'javascript'","computed":false}},"theme":{"type":{"name":"string"},"required":false,"description":"Theme of code, Default `\"idea\"`","defaultValue":{"value":"'monokai'","computed":false}},"codeLink":{"type":{"name":"string"},"required":false,"description":"Link to code, ex: CodeSandbox, CodePen. Default `\"\"`","defaultValue":{"value":"''","computed":false}},"buttonBgColor":{"type":{"name":"string"},"required":false,"description":"Background color of the buttons in the header. Default `\"dark\"`","defaultValue":{"value":"'dark'","computed":false}},"buttonBorderNone":{"type":{"name":"bool"},"required":false,"description":"Hide or show buttons border. Default `true`","defaultValue":{"value":"true","computed":false}},"buttonTextColor":{"type":{"name":"string"},"required":false,"description":"Color of the buttons in the header. Default `\"warning\"`","defaultValue":{"value":"'warning'","computed":false}},"buttonPaddingTop":{"type":{"name":"string"},"required":false,"description":"Paddin top of the buttons in the header. Default `\"0\"`","defaultValue":{"value":"'0'","computed":false}},"buttonPaddingBottom":{"type":{"name":"string"},"required":false,"description":"Paddin bottom of the buttons in the header. Default `\"0\"`","defaultValue":{"value":"'0'","computed":false}},"updateCode":{"type":{"name":"func"},"required":false,"description":"Function to handle the change made in the code.","defaultValue":{"value":"() => {}","computed":false}},"children":{"type":{"name":"union","value":[{"name":"arrayOf","value":{"name":"node"}},{"name":"node"}]},"required":false,"description":"Array of element to show inside Code","defaultValue":{"value":"null","computed":false}}},"filename":"/components/Code/Code.jsx"}