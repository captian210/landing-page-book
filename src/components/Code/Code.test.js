import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Code from './Code';

configure({ adapter: new Adapter() });

describe("Code", () => {
	let props;
	let mounted;
	const code = () => {
		if (!mounted) {
			mounted = mount(<Code {...props} />);
		}
		return mounted;
	};
	describe('Code', () => {

		beforeEach(() => {
			props = {};
			mounted = undefined;
		});

		it('should render', () => {
			expect(mount(<Code {...props} />)).toMatchSnapshot();
		});
		it("always renders a div", () => {
			const divs = code().find("div");
			expect(divs.length).toBeGreaterThan(0);
        });
        it("download", async () => {
			let res = await GithubService.getRepositoriesDownloads("chubin","chubin");
			expect(res);
		});
	});
})