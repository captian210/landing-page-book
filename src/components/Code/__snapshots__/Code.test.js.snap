// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Code should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Code
    bgColorDark={false}
    code="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
    codeLink=""
    collapsed={false}
    collapsible={false}
    languageCode="javascript"
    lineNumbers={true}
    readOnly={false}
    showCopyButton={true}
    showDeleteButton={true}
    showfooter={true}
    showheader={true}
    theme="idea"
    updateCode={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        <UNDEFINED>
          <Navbar
            brandLink=""
            brandLogo=""
            brandName=""
            className="CodeMirror__header"
            expand=""
            fixed={false}
            leftItems={Array []}
            onItemClick={[Function]}
            rightItems={Array []}
            transparent={false}
          >
            <NavbarNav
              alignItems="right"
            >
              
              <CopyToClipboard
                onCopy={[Function]}
                options={undefined}
                text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
              >
                <Button
                  active={false}
                  ariaLabel=""
                  circle={false}
                  className="btn  CodeMirror_btn"
                  color="default"
                  disabled={false}
                  loading={false}
                  loadingClass="fa fa-circle-o-notch fa-spin"
                  onClick={[Function]}
                  outline={false}
                  rounded={true}
                  size=""
                  style={Object {}}
                  tooltip="Copy"
                  type="button"
                >
                  <Icon
                    className=""
                    icon="fa fa-clone"
                    onClick={[Function]}
                    role="link"
                    style={Object {}}
                    title=""
                  />
                </Button>
              </CopyToClipboard>
              <Button
                active={false}
                ariaLabel=""
                circle={false}
                className="btn  CodeMirror_btn"
                color="default"
                disabled={false}
                loading={false}
                loadingClass="fa fa-circle-o-notch fa-spin"
                onClick={[Function]}
                outline={false}
                rounded={true}
                size=""
                style={Object {}}
                tooltip="Clear"
                type="button"
              >
                <Icon
                  className=""
                  icon="fa fa-trash-o"
                  onClick={[Function]}
                  style={Object {}}
                  title=""
                />
              </Button>
            </NavbarNav>
          </Navbar>
          <Unknown
            autoFocus={true}
            onChange={[Function]}
            options={
              Object {
                "lineNumbers": true,
                "mode": "javascript",
                "readOnly": false,
                "theme": "idea",
              }
            }
            preserveScrollPosition={false}
            value="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
          />
        </UNDEFINED>,
        false,
      ],
      "className": "CodeMirror__Container",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Navbar
              brandLink=""
              brandLogo=""
              brandName=""
              className="CodeMirror__header"
              expand=""
              fixed={false}
              leftItems={Array []}
              onItemClick={[Function]}
              rightItems={Array []}
              transparent={false}
            >
              <NavbarNav
                alignItems="right"
              >
                
                <CopyToClipboard
                  onCopy={[Function]}
                  options={undefined}
                  text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
                >
                  <Button
                    active={false}
                    ariaLabel=""
                    circle={false}
                    className="btn  CodeMirror_btn"
                    color="default"
                    disabled={false}
                    loading={false}
                    loadingClass="fa fa-circle-o-notch fa-spin"
                    onClick={[Function]}
                    outline={false}
                    rounded={true}
                    size=""
                    style={Object {}}
                    tooltip="Copy"
                    type="button"
                  >
                    <Icon
                      className=""
                      icon="fa fa-clone"
                      onClick={[Function]}
                      role="link"
                      style={Object {}}
                      title=""
                    />
                  </Button>
                </CopyToClipboard>
                <Button
                  active={false}
                  ariaLabel=""
                  circle={false}
                  className="btn  CodeMirror_btn"
                  color="default"
                  disabled={false}
                  loading={false}
                  loadingClass="fa fa-circle-o-notch fa-spin"
                  onClick={[Function]}
                  outline={false}
                  rounded={true}
                  size=""
                  style={Object {}}
                  tooltip="Clear"
                  type="button"
                >
                  <Icon
                    className=""
                    icon="fa fa-trash-o"
                    onClick={[Function]}
                    style={Object {}}
                    title=""
                  />
                </Button>
              </NavbarNav>
            </Navbar>,
            <Unknown
              autoFocus={true}
              onChange={[Function]}
              options={
                Object {
                  "lineNumbers": true,
                  "mode": "javascript",
                  "readOnly": false,
                  "theme": "idea",
                }
              }
              preserveScrollPosition={false}
              value="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "brandLink": "",
              "brandLogo": "",
              "brandName": "",
              "children": <NavbarNav
                alignItems="right"
              >
                
                <CopyToClipboard
                  onCopy={[Function]}
                  options={undefined}
                  text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
                >
                  <Button
                    active={false}
                    ariaLabel=""
                    circle={false}
                    className="btn  CodeMirror_btn"
                    color="default"
                    disabled={false}
                    loading={false}
                    loadingClass="fa fa-circle-o-notch fa-spin"
                    onClick={[Function]}
                    outline={false}
                    rounded={true}
                    size=""
                    style={Object {}}
                    tooltip="Copy"
                    type="button"
                  >
                    <Icon
                      className=""
                      icon="fa fa-clone"
                      onClick={[Function]}
                      role="link"
                      style={Object {}}
                      title=""
                    />
                  </Button>
                </CopyToClipboard>
                <Button
                  active={false}
                  ariaLabel=""
                  circle={false}
                  className="btn  CodeMirror_btn"
                  color="default"
                  disabled={false}
                  loading={false}
                  loadingClass="fa fa-circle-o-notch fa-spin"
                  onClick={[Function]}
                  outline={false}
                  rounded={true}
                  size=""
                  style={Object {}}
                  tooltip="Clear"
                  type="button"
                >
                  <Icon
                    className=""
                    icon="fa fa-trash-o"
                    onClick={[Function]}
                    style={Object {}}
                    title=""
                  />
                </Button>
              </NavbarNav>,
              "className": "CodeMirror__header",
              "expand": "",
              "fixed": false,
              "leftItems": Array [],
              "onItemClick": [Function],
              "rightItems": Array [],
              "transparent": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alignItems": "right",
                "children": Array [
                  "",
                  <CopyToClipboard
                    onCopy={[Function]}
                    options={undefined}
                    text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
                  >
                    <Button
                      active={false}
                      ariaLabel=""
                      circle={false}
                      className="btn  CodeMirror_btn"
                      color="default"
                      disabled={false}
                      loading={false}
                      loadingClass="fa fa-circle-o-notch fa-spin"
                      onClick={[Function]}
                      outline={false}
                      rounded={true}
                      size=""
                      style={Object {}}
                      tooltip="Copy"
                      type="button"
                    >
                      <Icon
                        className=""
                        icon="fa fa-clone"
                        onClick={[Function]}
                        role="link"
                        style={Object {}}
                        title=""
                      />
                    </Button>
                  </CopyToClipboard>,
                  <Button
                    active={false}
                    ariaLabel=""
                    circle={false}
                    className="btn  CodeMirror_btn"
                    color="default"
                    disabled={false}
                    loading={false}
                    loadingClass="fa fa-circle-o-notch fa-spin"
                    onClick={[Function]}
                    outline={false}
                    rounded={true}
                    size=""
                    style={Object {}}
                    tooltip="Clear"
                    type="button"
                  >
                    <Icon
                      className=""
                      icon="fa fa-trash-o"
                      onClick={[Function]}
                      style={Object {}}
                      title=""
                    />
                  </Button>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Button
                      active={false}
                      ariaLabel=""
                      circle={false}
                      className="btn  CodeMirror_btn"
                      color="default"
                      disabled={false}
                      loading={false}
                      loadingClass="fa fa-circle-o-notch fa-spin"
                      onClick={[Function]}
                      outline={false}
                      rounded={true}
                      size=""
                      style={Object {}}
                      tooltip="Copy"
                      type="button"
                    >
                      <Icon
                        className=""
                        icon="fa fa-clone"
                        onClick={[Function]}
                        role="link"
                        style={Object {}}
                        title=""
                      />
                    </Button>,
                    "onCopy": [Function],
                    "options": undefined,
                    "text": "const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "active": false,
                      "ariaLabel": "",
                      "children": <Icon
                        className=""
                        icon="fa fa-clone"
                        onClick={[Function]}
                        role="link"
                        style={Object {}}
                        title=""
                      />,
                      "circle": false,
                      "className": "btn  CodeMirror_btn",
                      "color": "default",
                      "disabled": false,
                      "loading": false,
                      "loadingClass": "fa fa-circle-o-notch fa-spin",
                      "onClick": [Function],
                      "outline": false,
                      "rounded": true,
                      "size": "",
                      "style": Object {},
                      "tooltip": "Copy",
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "className": "",
                        "icon": "fa fa-clone",
                        "onClick": [Function],
                        "role": "link",
                        "style": Object {},
                        "title": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "ariaLabel": "",
                    "children": <Icon
                      className=""
                      icon="fa fa-trash-o"
                      onClick={[Function]}
                      style={Object {}}
                      title=""
                    />,
                    "circle": false,
                    "className": "btn  CodeMirror_btn",
                    "color": "default",
                    "disabled": false,
                    "loading": false,
                    "loadingClass": "fa fa-circle-o-notch fa-spin",
                    "onClick": [Function],
                    "outline": false,
                    "rounded": true,
                    "size": "",
                    "style": Object {},
                    "tooltip": "Clear",
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "",
                      "icon": "fa fa-trash-o",
                      "onClick": [Function],
                      "style": Object {},
                      "title": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoFocus": true,
              "onChange": [Function],
              "options": Object {
                "lineNumbers": true,
                "mode": "javascript",
                "readOnly": false,
                "theme": "idea",
              },
              "preserveScrollPosition": false,
              "value": "const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};",
            },
            "ref": "editor",
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          <UNDEFINED>
            <Navbar
              brandLink=""
              brandLogo=""
              brandName=""
              className="CodeMirror__header"
              expand=""
              fixed={false}
              leftItems={Array []}
              onItemClick={[Function]}
              rightItems={Array []}
              transparent={false}
            >
              <NavbarNav
                alignItems="right"
              >
                
                <CopyToClipboard
                  onCopy={[Function]}
                  options={undefined}
                  text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
                >
                  <Button
                    active={false}
                    ariaLabel=""
                    circle={false}
                    className="btn  CodeMirror_btn"
                    color="default"
                    disabled={false}
                    loading={false}
                    loadingClass="fa fa-circle-o-notch fa-spin"
                    onClick={[Function]}
                    outline={false}
                    rounded={true}
                    size=""
                    style={Object {}}
                    tooltip="Copy"
                    type="button"
                  >
                    <Icon
                      className=""
                      icon="fa fa-clone"
                      onClick={[Function]}
                      role="link"
                      style={Object {}}
                      title=""
                    />
                  </Button>
                </CopyToClipboard>
                <Button
                  active={false}
                  ariaLabel=""
                  circle={false}
                  className="btn  CodeMirror_btn"
                  color="default"
                  disabled={false}
                  loading={false}
                  loadingClass="fa fa-circle-o-notch fa-spin"
                  onClick={[Function]}
                  outline={false}
                  rounded={true}
                  size=""
                  style={Object {}}
                  tooltip="Clear"
                  type="button"
                >
                  <Icon
                    className=""
                    icon="fa fa-trash-o"
                    onClick={[Function]}
                    style={Object {}}
                    title=""
                  />
                </Button>
              </NavbarNav>
            </Navbar>
            <Unknown
              autoFocus={true}
              onChange={[Function]}
              options={
                Object {
                  "lineNumbers": true,
                  "mode": "javascript",
                  "readOnly": false,
                  "theme": "idea",
                }
              }
              preserveScrollPosition={false}
              value="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
            />
          </UNDEFINED>,
          false,
        ],
        "className": "CodeMirror__Container",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Navbar
                brandLink=""
                brandLogo=""
                brandName=""
                className="CodeMirror__header"
                expand=""
                fixed={false}
                leftItems={Array []}
                onItemClick={[Function]}
                rightItems={Array []}
                transparent={false}
              >
                <NavbarNav
                  alignItems="right"
                >
                  
                  <CopyToClipboard
                    onCopy={[Function]}
                    options={undefined}
                    text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
                  >
                    <Button
                      active={false}
                      ariaLabel=""
                      circle={false}
                      className="btn  CodeMirror_btn"
                      color="default"
                      disabled={false}
                      loading={false}
                      loadingClass="fa fa-circle-o-notch fa-spin"
                      onClick={[Function]}
                      outline={false}
                      rounded={true}
                      size=""
                      style={Object {}}
                      tooltip="Copy"
                      type="button"
                    >
                      <Icon
                        className=""
                        icon="fa fa-clone"
                        onClick={[Function]}
                        role="link"
                        style={Object {}}
                        title=""
                      />
                    </Button>
                  </CopyToClipboard>
                  <Button
                    active={false}
                    ariaLabel=""
                    circle={false}
                    className="btn  CodeMirror_btn"
                    color="default"
                    disabled={false}
                    loading={false}
                    loadingClass="fa fa-circle-o-notch fa-spin"
                    onClick={[Function]}
                    outline={false}
                    rounded={true}
                    size=""
                    style={Object {}}
                    tooltip="Clear"
                    type="button"
                  >
                    <Icon
                      className=""
                      icon="fa fa-trash-o"
                      onClick={[Function]}
                      style={Object {}}
                      title=""
                    />
                  </Button>
                </NavbarNav>
              </Navbar>,
              <Unknown
                autoFocus={true}
                onChange={[Function]}
                options={
                  Object {
                    "lineNumbers": true,
                    "mode": "javascript",
                    "readOnly": false,
                    "theme": "idea",
                  }
                }
                preserveScrollPosition={false}
                value="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "brandLink": "",
                "brandLogo": "",
                "brandName": "",
                "children": <NavbarNav
                  alignItems="right"
                >
                  
                  <CopyToClipboard
                    onCopy={[Function]}
                    options={undefined}
                    text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
                  >
                    <Button
                      active={false}
                      ariaLabel=""
                      circle={false}
                      className="btn  CodeMirror_btn"
                      color="default"
                      disabled={false}
                      loading={false}
                      loadingClass="fa fa-circle-o-notch fa-spin"
                      onClick={[Function]}
                      outline={false}
                      rounded={true}
                      size=""
                      style={Object {}}
                      tooltip="Copy"
                      type="button"
                    >
                      <Icon
                        className=""
                        icon="fa fa-clone"
                        onClick={[Function]}
                        role="link"
                        style={Object {}}
                        title=""
                      />
                    </Button>
                  </CopyToClipboard>
                  <Button
                    active={false}
                    ariaLabel=""
                    circle={false}
                    className="btn  CodeMirror_btn"
                    color="default"
                    disabled={false}
                    loading={false}
                    loadingClass="fa fa-circle-o-notch fa-spin"
                    onClick={[Function]}
                    outline={false}
                    rounded={true}
                    size=""
                    style={Object {}}
                    tooltip="Clear"
                    type="button"
                  >
                    <Icon
                      className=""
                      icon="fa fa-trash-o"
                      onClick={[Function]}
                      style={Object {}}
                      title=""
                    />
                  </Button>
                </NavbarNav>,
                "className": "CodeMirror__header",
                "expand": "",
                "fixed": false,
                "leftItems": Array [],
                "onItemClick": [Function],
                "rightItems": Array [],
                "transparent": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "alignItems": "right",
                  "children": Array [
                    "",
                    <CopyToClipboard
                      onCopy={[Function]}
                      options={undefined}
                      text="const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};"
                    >
                      <Button
                        active={false}
                        ariaLabel=""
                        circle={false}
                        className="btn  CodeMirror_btn"
                        color="default"
                        disabled={false}
                        loading={false}
                        loadingClass="fa fa-circle-o-notch fa-spin"
                        onClick={[Function]}
                        outline={false}
                        rounded={true}
                        size=""
                        style={Object {}}
                        tooltip="Copy"
                        type="button"
                      >
                        <Icon
                          className=""
                          icon="fa fa-clone"
                          onClick={[Function]}
                          role="link"
                          style={Object {}}
                          title=""
                        />
                      </Button>
                    </CopyToClipboard>,
                    <Button
                      active={false}
                      ariaLabel=""
                      circle={false}
                      className="btn  CodeMirror_btn"
                      color="default"
                      disabled={false}
                      loading={false}
                      loadingClass="fa fa-circle-o-notch fa-spin"
                      onClick={[Function]}
                      outline={false}
                      rounded={true}
                      size=""
                      style={Object {}}
                      tooltip="Clear"
                      type="button"
                    >
                      <Icon
                        className=""
                        icon="fa fa-trash-o"
                        onClick={[Function]}
                        style={Object {}}
                        title=""
                      />
                    </Button>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Button
                        active={false}
                        ariaLabel=""
                        circle={false}
                        className="btn  CodeMirror_btn"
                        color="default"
                        disabled={false}
                        loading={false}
                        loadingClass="fa fa-circle-o-notch fa-spin"
                        onClick={[Function]}
                        outline={false}
                        rounded={true}
                        size=""
                        style={Object {}}
                        tooltip="Copy"
                        type="button"
                      >
                        <Icon
                          className=""
                          icon="fa fa-clone"
                          onClick={[Function]}
                          role="link"
                          style={Object {}}
                          title=""
                        />
                      </Button>,
                      "onCopy": [Function],
                      "options": undefined,
                      "text": "const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "active": false,
                        "ariaLabel": "",
                        "children": <Icon
                          className=""
                          icon="fa fa-clone"
                          onClick={[Function]}
                          role="link"
                          style={Object {}}
                          title=""
                        />,
                        "circle": false,
                        "className": "btn  CodeMirror_btn",
                        "color": "default",
                        "disabled": false,
                        "loading": false,
                        "loadingClass": "fa fa-circle-o-notch fa-spin",
                        "onClick": [Function],
                        "outline": false,
                        "rounded": true,
                        "size": "",
                        "style": Object {},
                        "tooltip": "Copy",
                        "type": "button",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {
                          "className": "",
                          "icon": "fa fa-clone",
                          "onClick": [Function],
                          "role": "link",
                          "style": Object {},
                          "title": "",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "active": false,
                      "ariaLabel": "",
                      "children": <Icon
                        className=""
                        icon="fa fa-trash-o"
                        onClick={[Function]}
                        style={Object {}}
                        title=""
                      />,
                      "circle": false,
                      "className": "btn  CodeMirror_btn",
                      "color": "default",
                      "disabled": false,
                      "loading": false,
                      "loadingClass": "fa fa-circle-o-notch fa-spin",
                      "onClick": [Function],
                      "outline": false,
                      "rounded": true,
                      "size": "",
                      "style": Object {},
                      "tooltip": "Clear",
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "className": "",
                        "icon": "fa fa-trash-o",
                        "onClick": [Function],
                        "style": Object {},
                        "title": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": true,
                "onChange": [Function],
                "options": Object {
                  "lineNumbers": true,
                  "mode": "javascript",
                  "readOnly": false,
                  "theme": "idea",
                },
                "preserveScrollPosition": false,
                "value": "const component = {
	name: \\"react-code\\",
	author: \\"front10-devs\\",
	repo: \\"https://gitlab.com/front10-devs/landing-page-book\\"
};",
              },
              "ref": "editor",
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
